cf-custom-resources/lib/dns-cert-validator.js

```diff
@@ -274,16 +368,26 @@ const updateRecords = function (
     .promise();
 };
 
-const clients = function (region) {
+const clients = function (region, rootDnsRole) {
   const acm = new aws.ACM({
     region,
   });
-  const route53 = new aws.Route53();
+  const envRoute53 = new aws.Route53();
+  if (waiter) {
+    // Used by the test suite, since waiters aren't mockable yet
+    envRoute53.waitFor = acm.waitFor = waiter;
+  }
```

Comment:
nit: I think we can remove one of these waiters and just set all of them together:
```suggestion
  if (waiter) {
    // Used by the test suite, since waiters aren't mockable yet
    appRoute53.waitFor = envRoute53.waitFor = acm.waitFor = waiter;
  }
```